### PR TITLE
fix: List pagination is null should not crash

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

https://codesandbox.io/s/p52mm2vj97

### 💡 Solution

`pagination && typeof pagination === 'object'`

### 📝 Changelog

Fix List `pagination` crash when is `null`

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by ikobe** [comment]
+1

File: components/list/index.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export default class List<T> extends React.Component<ListProps<T>, ListState> {
     super(props);
 
     const { pagination } = props;
-    const paginationObj = typeof pagination === 'object' ? pagination : {};
+    const paginationObj = pagination && typeof pagination === 'object' ? pagination : {};
```
Comment (fengmk2) [inline components/list/index.tsx]: 不是都 ts 了么。。。给 pagination 加个类型校验，就不需要这样写了。`pagination  || {}` 就好了。

Comment (afc163) [reply components/list/index.tsx]: 不行的，pagination 为 `false` 和 `null` 时应该是不展示分页的，也是默认逻辑。

image

Comment (zombieJ) [reply components/list/index.tsx]: 这里 `pagination` 的定义是: `pagination?: PaginationConfig | false;`，ts 里跑是会报错的：
屏幕快照 2019-04-22 下午5 54 56

Comment (zombieJ) [reply components/list/index.tsx]: 这边是初始化逻辑，展示逻辑在下面会检查。

**Pull request merged**